java/src/main/native/src/AggregationJni.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/aggregation.hpp>
+
+#include "cudf_jni_apis.hpp"
+
+extern "C" {
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_Aggregation_close(JNIEnv *env,
+                                                             jclass class_object,
+                                                             jlong ptr) {
+  try {
+    cudf::jni::auto_set_device(env);
+    // Make sure that we close this...
+    auto to_del = reinterpret_cast<std::unique_ptr<cudf::aggregation> *>(ptr);
+    delete to_del;
+  }
+  CATCH_STD(env, );
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Aggregation_createNoParamAgg(JNIEnv *env,
+                                                                         jclass class_object,
+                                                                         jint kind) {
+  try {
+    cudf::jni::auto_set_device(env);
+    std::unique_ptr<std::unique_ptr<cudf::aggregation>> ret(
```

Comment:
Having two-levels of `unique_ptr` here seems a bit odd.  Do we need to have Java track a pointer to `std::unique_ptr<cudf::aggregation>`?  It appears to be simpler to track a pointer to `cudf::aggregation` directly/  The rest of the JNI code unwraps the `unique_ptr` when Java tracks something, wondering why we shouldn't do the same here.  I think it would reduce a level of indirection where it's used as well, where we're calling the aggregation's `clone()` method anyway to produce the value ultimately passed to the cudf method.